File: backend/Models/Reservation.js
const { Model, DataTypes } = require('sequelize');
const sequelize = require('../config/dbConfig');
const TripOffer = require('./TripOffer');  // Assurez-vous que le modèle TripOffer est correctement importé

class Reservation extends Model {}

Reservation.init({
    id: {
        type: DataTypes.INTEGER,
        allowNull: false,
        autoIncrement: true,
        unique: true,
        primaryKey: true
    },
    reservation_date: {
        type: DataTypes.DATE,
        allowNull: false
    },
    reservation_seats: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    status: {
        type: DataTypes.BOOLEAN,
        allowNull: false
    },
    total_price: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    trip_offer: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: 'TripOffers',  
            key: 'id'
        },
        onUpdate: 'CASCADE',
        onDelete: 'SET NULL'
    },
    user_id:{
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: 'users',  
            key: 'id'
        },
        onUpdate: 'CASCADE',
        onDelete: 'SET NULL'   
    },
    createdAt: {
        allowNull: false,
        type: DataTypes.DATE
    },
    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE
    }
}, {
    sequelize,
    modelName: 'Reservation'
});

Reservation.belongsTo(TripOffer, { as: 'offer', foreignKey: 'trip_offer' });


module.exports = Reservation;

File: frontend/appMobile/untitled/lib/classes/reservation.dart
class Reservation {
  final int id;
  final String reservationDate;
  final int reservationSeats;
  final bool status;
  final int totalPrice;
  final int tripOfferId;
  final int user_id;

  Reservation({
    required this.id,
    required this.reservationDate,
    required this.reservationSeats,
    required this.status,
    required this.totalPrice,
    required this.tripOfferId,
    required this.user_id
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'] ?? 0, // Default to 0 if null
      reservationDate: json['reservation_date'] ?? '', // Default to empty string if null
      reservationSeats: json['reservation_seats'] ?? 0, // Default to 0 if null
      status: json['status'] ?? false, // Default to false if null
      totalPrice: json['total_price'] ?? 0, // Default to 0 if null
      tripOfferId: json['trip_offer'] ?? 0, // Default to 0 if null
      user_id: json['user_id'] ?? 0, // Default to 0 if null

    );
  }
}
